src/time/src/mcp_server_time/server-resources.py

```diff
@@ -0,0 +1,167 @@
+            },
+            "time_difference": f"{(target_time.utcoffset() - source_time.utcoffset()).total_seconds() / 3600:+.1f}h",
+            "date_changed": date_changed,
+            "day_relation": "next day" if date_changed and target_time.date() > source_time.date() else "previous day" if date_changed else "same day"
+        }
+
+    async def handle_uri(self, uri: str) -> str:
+        """Main URI handler that routes to appropriate method"""
+        try:
+            if uri == "time://query":
+                result = self.get_current_time()
+            elif uri.startswith("time://query/"):
+                timezone_name = uri.replace("time://query/", "")
+                result = self.get_current_time(timezone_name)
+            elif uri.startswith("time://convert/"):
+                path = uri.replace("time://convert/", "")
+                source_parts = path.split("/to/")
+                if len(source_parts) != 2:
+                    raise ValueError("Invalid conversion URI format")
+                
+                source_info, target_tz = source_parts
+                source_parts = source_info.split("/")
+                if len(source_parts) != 2:
+                    raise ValueError("Invalid source time format")
+                
+                source_tz, time_str = source_parts
+                result = self.convert_time(source_tz, time_str, target_tz)
+            else:
+                raise ValueError(f"Unsupported URI format: {uri}")
+                
+            return json.dumps(result)
+            
+        except Exception as e:
+            raise ValueError(f"Error processing time query: {str(e)}")
+
+
+async def serve(local_timezone: Optional[str] = None) -> None:
+    server = Server("mcp-time")
+    time_server = TimeServer(local_timezone)
+
+    @server.list_resources()
+    async def list_resources() -> list[Resource]:
+        current_tz = str(time_server.local_tz)
+
+        return [
+            Resource(
+                uri=AnyUrl("time://query"),
+                name="Local Time Query",
+                description=f"Get current time in your local timezone ({current_tz})",
+                mimeType="application/json"
+            )
+        ]
+            # resourceTemplates=[
+            #     ResourceTemplate(
+            #         uriTemplate="time://query/{timezone}",
+            #         name="Timezone Time Query",
+            #         description="Get current time in a specific timezone",
+            #         mimeType="application/json"
+            #     ),
+            #     ResourceTemplate(
+            #         uriTemplate="time://convert/{source_timezone}/{hour}:{minute}/to/{target_timezone}",
+            #         name="Timezone Conversion",
+            #         description="Convert specific time between timezones (use 'local' for local timezone)",
+            #         mimeType="application/json"
+            #     )
+            # ]
```

Comment:
I'm guessing this is because of https://github.com/modelcontextprotocol/python-sdk/issues/68? Sorry about that. Do you mind adding a comment to that effect, so it's clear this bit can be fixed up once that issue is resolved?

Reply:
"resource-server" was the implementation's first version, which I accidentally added to the repo.

In the first implementation, I wanted to use Resources and ResourceTemplates in order to LLM call this resources without permission from the user, but I've noticed that Claude's desktop client doesn't run this time Resources at all...

Then, I noticed in the docs that calling Resources is fully managed by the client app itself, and because I never worked for Claude's desktop during my tests, I switched to the Tools usage.

@jspahrsummers If you like the idea of having both tools and resources with the same functionalities, I will add Resources to the main 'server.py' again with proper comment about ResourceTemplate

Then, someone can use time Resources in the client application